src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,23 @@ mod bar {
 ```
 "##,
 
+E0253: r##"
+Attempt was made to import an unimportable value.  This can happen when
+trying to import a function from a trait.  An example of this error:
+
+```
+mod foo {
+    pub trait MyTrait {
+        fn doSomething();
+    }
+}
+use foo::MyTrait::doSomething;
+```
+
+In general, it's not legal to directly import functions from a crate or
```

Comment:
agreed.

make sure you make the methods vs functions thing clear, for many people those are the same thing.